workflow/snakemake_rules/core.smk

```diff
@@ -41,28 +40,70 @@ rule filter:
         sequences="data/sequences.fasta",
         metadata="results/metadata.tsv",
         exclude=config["exclude"],
+    output:
+        sequences=build_dir + "/{build_name}/good_sequences.fasta",
+        metadata=build_dir + "/{build_name}/good_metadata.tsv",
+        log=build_dir + "/{build_name}/good_filter.log",
+    params:
+        min_date=config["min_date"],
+        min_length=config["min_length"],
+    shell:
+        """
+        augur filter \
+            --sequences {input.sequences} \
+            --metadata {input.metadata} \
+            --exclude {input.exclude} \
+            --output-sequences {output.sequences} \
+            --output-metadata {output.metadata} \
+            --min-date {params.min_date} \
+            --min-length {params.min_length} \
+            --output-log {output.log}
+        """
+
+
+rule include_A_strains:
+    input:
+        metadata=rules.exclude_bad.output.metadata,
+    output:
+        include_strains=build_dir + "/{build_name}/include_strains.txt",
+    shell:
+        """
+        tsv-filter -H \
+            --str-in-fld lineage:'A'\
+            {input.metadata} \
+        | tsv-select -f 1 \
+        > {output.include_strains}
+        """
+
+
+rule filter:
+    message:
+        """
+        Filtering to
+          - {params.sequences_per_group} sequence(s) per {params.group_by!s}
+        """
+    input:
+        sequences=rules.exclude_bad.output.sequences,
+        metadata=rules.exclude_bad.output.metadata,
+        include=rules.include_A_strains.output.include_strains,
```

Comment:
It would be easier for people to add their own custom rules if we keep all of inputs as the file path rather than the rule outputs.